server/src/main/java/org/elasticsearch/gateway/LocalAllocateDangledIndices.java

```diff
@@ -176,7 +175,10 @@ public ClusterState execute(ClusterState currentState) {
                         try {
                             // The dangled index might be from an older version, we need to make sure it's compatible
                             // with the current version.
-                            newIndexMetadata = indexMetadataVerifier.verifyIndexMetadata(indexMetadata, minIndexCompatibilityVersion);
+                            newIndexMetadata = indexMetadataVerifier.verifyIndexMetadata(
```

Comment:
verifyIndexMetadata should be updated to take IndexVersion (in a followup)?

Reply:
Yes, index metadata is on the list for updating - but it needs access to index versions in cluster state first